java/address-java/src/test/java/uk/os/elements/PostcodeUtilTest.java

```diff
@@ -0,0 +1,443 @@
+    }
+
+    @Test
+    public void shouldNotMatchOtherThanArea() {
+        checkIsLikelyAreaPostcodeAndIgnoreCaseAndSpacing(false, DISTRICT_POSTCODES);
+        checkIsLikelyAreaPostcodeAndIgnoreCaseAndSpacing(false, SECTOR_POSTCODES);
+        checkIsLikelyAreaPostcodeAndIgnoreCaseAndSpacing(false, UNIT_POSTCODES);
+        checkIsLikelyAreaPostcodeAndIgnoreCaseAndSpacing(false, PLACE_ORDNANCE_SURVEY, PLACE_BBC, PLACE_BUCKINGHAM_PALACE,
+                PLACE_THREE_CHAR, PLACE_FOUR_CHAR );
+    }
+
+    @Test
+    public void shouldNotMatchOtherThanUnit() {
+        checkIsLikelyFullPostcodeAndIgnoreCaseAndSpacing(false, AREA_POSTCODES);
+        checkIsLikelyFullPostcodeAndIgnoreCaseAndSpacing(false, AREA_POSTCODES);
+        checkIsLikelyFullPostcodeAndIgnoreCaseAndSpacing(false, DISTRICT_POSTCODES);
+        checkIsLikelyFullPostcodeAndIgnoreCaseAndSpacing(false, SECTOR_POSTCODES);
+        checkIsLikelyFullPostcodeAndIgnoreCaseAndSpacing(false, ALTERNATIVES);
+    }
+
+    @Test
+    public void shouldNotMatchPlacesWithFullOrLikelyPostcode() {
+        assertFalse(PostcodeUtil.isLikelyFullPostcode(PLACE_ORDNANCE_SURVEY));
+        assertFalse(PostcodeUtil.isLikelyFullPostcode(PLACE_BBC));
+        assertFalse(PostcodeUtil.isLikelyFullPostcode(PLACE_BUCKINGHAM_PALACE));
+        assertFalse(PostcodeUtil.isLikelyFullPostcode(PLACE_ONE_CHAR));
+        assertFalse(PostcodeUtil.isLikelyFullPostcode(PLACE_TWO_CHAR));
+        assertFalse(PostcodeUtil.isLikelyFullPostcode(PLACE_THREE_CHAR));
+        assertFalse(PostcodeUtil.isLikelyFullPostcode(PLACE_FOUR_CHAR));
+
+
+        assertFalse(PostcodeUtil.isLikelyPostcode(PLACE_ORDNANCE_SURVEY));
+        assertFalse(PostcodeUtil.isLikelyPostcode(PLACE_BBC));
+        assertFalse(PostcodeUtil.isLikelyPostcode(PLACE_BUCKINGHAM_PALACE));
+        assertFalse(PostcodeUtil.isLikelyPostcode(PLACE_ONE_CHAR));
+        assertFalse(PostcodeUtil.isLikelyPostcode(PLACE_TWO_CHAR));
+        assertFalse(PostcodeUtil.isLikelyPostcode(PLACE_THREE_CHAR));
+        assertFalse(PostcodeUtil.isLikelyPostcode(PLACE_FOUR_CHAR));
+    }
+
+    @Test
+    public void shouldNotMatchUnitPostcodesStrict() {
+        checkIsLikelyUnitPostcodeStrictAndIgnoreCase(false, AREA_POSTCODES);
+        checkIsLikelyUnitPostcodeStrictAndIgnoreCase(false, DISTRICT_POSTCODES);
+        checkIsLikelyUnitPostcodeStrictAndIgnoreCase(false, SECTOR_POSTCODES);
+        checkIsLikelyUnitPostcodeStrictAndIgnoreCase(false, withoutSpaces(UNIT_POSTCODES));
+        checkIsLikelyUnitPostcodeStrictAndIgnoreCase(false, ALTERNATIVES);
+    }
+
+    private void checkIsLikelyFullPostcodeAndIgnoreCaseAndSpacing(String s) {
+        checkIsLikelyFullPostcodeAndIgnoreCaseAndSpacing(true, s);
+    }
+
+    private void checkIsLikelyFullPostcodeAndIgnoreCaseAndSpacing(boolean isExpected, String... strings) {
+        for (String s : strings) {
+            if (isExpected) {
+                assertTrue("test input", PostcodeUtil.isLikelyFullPostcode(s));
```

Comment:
Could DRY this out a little bit and lose the `if`/`else` if you just did `assertTrue("test input", PostcodeUtil.isLikelyFullPostcode(s) == isExpected);`